modules/features/podcasts/src/main/java/au/com/shiftyjelly/pocketcasts/podcasts/viewmodel/GiveRatingViewModel.kt

```diff
@@ -85,38 +79,46 @@ class GiveRatingViewModel @Inject constructor(
                     } else {
-                        val podcast = podcastManager.findPodcastByUuidSuspend(podcastUuid)
-                        if (podcast == null) {
-                            _state.value = State.ErrorWhenLoadingPodcast
-                        } else {
-                            val rating = ratingsManager.podcastRatings(podcastUuid).first()
-                            val stars: Stars? = getStarsFromRating(rating)
-
-                            _state.value = State.Loaded(
-                                podcastUuid = podcast.uuid,
-                                podcastTitle = podcast.title,
-                                _stars = stars,
-                            )
-                        }
+                        onSuccess()
                     }
                 }
             }
         }
     }
 
-    private fun getStarsFromRating(podcastRatings: PodcastRatings): Stars? {
-        val rating = podcastRatings.average ?: 0.0
-
-        val ratingInt = rating.toInt()
-        val half = rating % 1 >= 0.5
-
-        return when (ratingInt) {
-            0 -> Stars.Zero
-            1 -> if (half) Stars.OneAndHalf else Stars.One
-            2 -> if (half) Stars.TwoAndHalf else Stars.Two
-            3 -> if (half) Stars.ThreeAndHalf else Stars.Three
-            4 -> if (half) Stars.FourAndHalf else Stars.Four
-            else -> Stars.Five // if the rating is somehow higher than 5, just return 5
+    fun loadData(podcastUuid: String) {
+        viewModelScope.launch {
+            _state.value = State.Loading
+
+            val podcast = podcastManager.findPodcastByUuidSuspend(podcastUuid)
+
+            if (podcast == null) {
+                _state.value = State.ErrorWhenLoadingPodcast
+            } else {
+                try {
+                    val ratedPodcast = syncManager.getPodcastRating(PodcastRatingShowRequest(podcastUuid))
+
+                    _state.value = State.Loaded(
+                        podcastUuid = podcast.uuid,
+                        podcastTitle = podcast.title,
+                        _currentSelectedRate = ratingToStars(ratedPodcast.podcastRating.toDouble()),
+                        previousRate = ratingToStars(ratedPodcast.podcastRating.toDouble()),
+                    )
+                } catch (e: HttpException) {
+                    if (e.code() == 404) { // Does not have previous rate
+                        _state.value = State.Loaded(
+                            podcastUuid = podcast.uuid,
+                            podcastTitle = podcast.title,
+                            _currentSelectedRate = null,
+                            previousRate = null,
+                        )
+                    } else {
+                        _state.value = State.ErrorWhenLoadingPodcast
+                    }
+                } catch (e: Exception) {
+                    _state.value = State.ErrorWhenLoadingPodcast
```

Comment:
Can we please log this exception?

Reply:
Sure! I also updated other areas with logs [bc8b3d2](https://github.com/Automattic/pocket-casts-android/pull/2483/commits/bc8b3d20171266f471bd9861b43954d22ed37e43)